### ISSUE TITLE
[Feature] Elasticsearch: update to new Java API

### ISSUE THREAD

**Issue by jloisel** [jobrunr/jobrunr#499]
Elasticsearch Rest High Level Client is deprecated in favor of:
https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/7.17/index.html

There is still the possibility to have both clients inside a single application:
https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/7.17/migrate-hlrc.html

Still, would be better to migrate to the new client which is the standard for Elasticsearch 8.x.x as well.

**Comment by jloisel** [comment]
Thanks for the feedback. As a workaround, we did:

- Add older sdk to classpath:

```xml
    <dependency>
      <groupId>org.elasticsearch.client</groupId>
      <artifactId>elasticsearch-rest-high-level-client</artifactId>
      <version>${elasticsearch.version}</version>
    </dependency>
```
- Create a Spring configuration to load the `RestHighLevelClient`:

```java
@Configuration
@ConditionalOnProperty(prefix = "org.jobrunr.job-scheduler", name = "enabled", havingValue = "true", matchIfMissing = true)
class JobRunrConfig {

  @Bean
  @SuppressWarnings("deprecated")
  RestHighLevelClient restHighLevelClient(final RestClient client) {
    return new RestHighLevelClientBuilder(client).build();
  }
}
```

**Comment by jloisel** [comment]
We currently use Quartz along with a postgres database. We already have Elasticsearch to store the majority of our data. By switching from Quartz to JobRunr, we can get rid of quartz and the postgresql database.

**Comment by rdehuyss** [comment]
I must confess that Elastic currently is not used by many installations in production - Most people use Postgres.

**Comment by rdehuyss** [comment]
Solved in branch 6.4.
